### DISCUSSION TITLE
Zoom functionality

### DISCUSSION THREAD

**Discussion by Melynt3** [capawesome-team/capacitor-mlkit#63] [General]
Hi!

I was just curious if  you are u planning on adding zoom feature in near future? 

By the way, the library is compatible with ionic 7?
Thank you very much in advance for the information, you are doing a great job!

 o/

**Comment by robingenz** [comment]
Thank you for your interest!

> I was just curious if you are u planning on adding zoom feature in near future?

It is not currently planned, but you are welcome to create a feature request.

> By the way, the library is compatible with ionic 7?

Yes!

**Comment by Melynt3** [comment]
That's great and i'll do make a request. 


Btw. i was trying to run it using ur [guide](https://ionic.io/blog/how-to-build-an-ionic-barcode-scanner-with-capacitor), but unforntunatly 

> ERROR Error: Uncaught (in promise): Error: The Google Barcode Scanner Module is not available. You must install it first.
> Error: The Google Barcode Scanner Module is not available. You must install it first.

Not really sure what does this mean, i installed evertthing and gave the permissions, any idea?

**Reply by Melynt3** [reply]
Okey, i found out that there is a demo and scan option works just fine, but the startScan in modal option throws:

I have all permissions.

I have opened modal but nothing else happens, and there's this error.

> main.6088e6590425bdd3.js:1 ERROR Error: Uncaught (in promise): Error: Attempt to invoke virtual method 'void androidx.camera.view.PreviewView.setScaleType(androidx.camera.view.PreviewView$ScaleType)' on a null object reference
> Error: Attempt to invoke virtual method 'void androidx.camera.view.PreviewView.setScaleType(androidx.camera.view.PreviewView$ScaleType)' on a null object reference


Couldnt find any information on this, halp

**Reply by robingenz** [reply]
Maybe this helps you: https://github.com/capawesome-team/capacitor-mlkit/issues/53#issuecomment-1611334676
I would need a [Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example) using [this template](https://github.com/capawesome-team/.capacitor-app) in a public GitHub repository to debug the issue since it works fine in my demo app.

**Reply by Melynt3** [reply]
It looks similar to my problem, but i dont use any iOS libraries that may colide like in #53 

Demo works fine but it have problem with code128 scanning, and i even tried to adapt ur implementation to my project but still; the same error occurs. 

What im doin is `ionic cap sync && ionic cap build android` and im debuging on my phone with android 12, is that what u were asking for?

**Reply by robingenz** [reply]
>  is that what u were asking for?

No, i need a [Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example) using [this template](https://github.com/capawesome-team/.capacitor-app) in a public GitHub repository. Please take a look at the linked guide and repository.

**Reply by Melynt3** [reply]
It might be a problem, cuz i cant reproduce it at minimal (and also, cant public the code, nda). 

Damn.. ill prolly have to check the dependencies in this case.
If you have a suspicion of where to start checking would be very grateful.

**Reply by Melynt3** [reply]
Sir, i managed to run it with my UI, but the another problem jump out, basically this is scanner for code128 format but i cant scan them because they are too small 😅 I know already that there's no plan for zoom, but do u know if i can zoom camera from the start? Or maybe open in macro mode?

Bigger one are no problem.  

thanks in advance 

best regards

**Reply by robingenz** [reply]
> I know already that there's no plan for zoom, but do u know if i can zoom camera from the start? Or maybe open in macro mode?

No, this is not yet possible. Please create a feature request.

**Comment by Melynt3** [comment]
@robingenz sorry for spam but i feel like hitting a wall for about a day already. 

Was trying even implement the whole part of demo and its still the same error, maybe that dependencies problem but im not sure of anything...

> "dependencies": {
> 		"@angular/common": "^16.1.1",
> 		"@angular/core": "^16.1.1",
> 		"@angular/forms": "^16.1.1",
> 		"@angular/platform-browser": "^16.1.1",
> 		"@angular/platform-browser-dynamic": "^16.1.1",
> 		"@angular/router": "^16.1.1",
> 		"@capacitor-community/bluetooth-le": "^3.0.0",
> 		"@capacitor-community/electron": "^4.1.2",
> 		"@capacitor-mlkit/barcode-scanning": "^5.0.3",
> 		"@capacitor/android": "5.0.5",
> 		"@capacitor/app": "5.0.3",
> 		"@capacitor/camera": "^5.0.4",
> 		"@capacitor/clipboard": "^5.0.6",
> 		"@capacitor/core": "5.0.5",
> 		"@capacitor/filesystem": "^5.0.4",
> 		"@capacitor/haptics": "5.0.4",
> 		"@capacitor/keyboard": "5.0.4",
> 		"@capacitor/network": "^5.0.4",
> 		"@capacitor/preferences": "^5.0.4",
> 		"@capacitor/splash-screen": "^5.0.4",
> 		"@capacitor/status-bar": "5.0.4",
> 		"@capawesome/capacitor-file-picker": "^5.1.0",
> 		"@electron/remote": "^2.0.9",
> 		"@ionic/angular": "7.1.0",
> 		"@ng-select/ng-select": "^11.0.0",
> 		"capacitor-resources": "^2.0.5",
> 		"crc": "^4.3.1",
> 		"datetime-next": "^1.2.6",
> 		"electron": "^25.1.1",
> 		"gauss-event": "^3.1.0",
> 		"gauss-front-core": "^3.1.3",
> 		"hydrator-next": "^1.4.3",
> 		"ionicons": "^7.1.2",
> 		"md5": "^2.3.0",
> 		"ngx-electron": "^2.2.0",
> 		"rxjs": "~7.5.0",
> 		"rxjs-extended": "^1.0.1",
> 		"socket.io-client": "^4.5.1",
> 		"stream": "^0.0.2",
> 		"td-byte-util": "^1.0.0",
> 		"td-number-util": "^1.0.0",
> 		"td-object-util": "^1.0.1",
> 		"td-serializer": "^1.0.2",
> 		"td-string-util": "^1.0.0",
> 		"td-time-util": "^1.0.0",
> 		"td-type-util": "^1.0.0",
> 		"td-zip": "^1.0.8",
> 		"tslib": "^2.3.0",
> 		"util": "^0.12.5",
> 		"zone.js": "~0.13.1"
> 	},
> 	"devDependencies": {
> 		"@angular-devkit/build-angular": "^16.1.0",
> 		"@angular-eslint/builder": "^14.0.0",
> 		"@angular-eslint/eslint-plugin": "^14.0.0",
> 		"@angular-eslint/eslint-plugin-template": "^14.0.0",
> 		"@angular-eslint/template-parser": "^14.0.0",
> 		"@angular/cli": "^16.1.0",
> 		"@angular/compiler": "^16.1.1",
> 		"@angular/compiler-cli": "^16.1.1",
> 		"@angular/language-service": "^16.1.1",
> 		"@capacitor/cli": "5.0.5",
> 		"@ionic/angular-toolkit": "9",
> 		"@ionic/lab": "3.2.13",
> 		"@types/crc": "^3.8.0",
> 		"@types/md5": "^2.3.2",
> 		"@types/node": "^12.11.1",
> 		"@types/web-bluetooth": "^0.0.16",
> 		"@typescript-eslint/eslint-plugin": "5.3.0",
> 		"@typescript-eslint/parser": "5.3.0",
> 		"electron-builder": "^24.4.0",
> 		"eslint": "^7.6.0",
> 		"eslint-plugin-import": "2.22.1",
> 		"eslint-plugin-jsdoc": "30.7.6",
> 		"eslint-plugin-prefer-arrow": "1.2.2",
> 		"ts-node": "~8.3.0",
> 		"typescript": "~5.1.3",
> 		"util": "^0.12.5"
> 	},
>
